cranelift/native/src/lib.rs

```diff
@@ -125,6 +125,26 @@ pub fn builder_with_options(
         }
     }
 
+    // There is no is_s390x_feature_detected macro yet, so for now
+    // we use getauxval from the libc crate directly.
+    #[cfg(all(target_arch = "s390x", target_os = "linux"))]
+    {
+        use cranelift_codegen::settings::Configurable;
+
+        if !infer_native_flags {
+            return Ok(isa_builder);
+        }
+
+        let v = unsafe { libc::getauxval(libc::AT_HWCAP) };
```

Comment:
It's too bad this isn't in the stdlib yet -- are there plans to try to upstream this? It looks like the [`std_detect` crate] in the stdlib has [similar implementations](https://github.com/rust-lang/stdarch/tree/master/crates/std_detect/src/detect) that use the ELF auxvec capability word(s) on other architectures.

Reply:
Apart from `is_x86_feature_detected!` I believe the target detection macros are all unstable anyway.

Reply:
Agreed, in the long run we'll want to have this in stdlib like the other platforms.